### PR TITLE
Survey summary page

### PR DESCRIPTION
**Pull request by leohtkam**

### CODE REVIEW

File: app/assets/html/survey_questions_edit.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<div id="survey-questions-container">
+    <div class="header">{{ currentTopic }}</div>
+    <div class="spacing"></div>
+    <div id="topic-questions-container">
+        <div id="question{{$index+1}}" class="question-container" ng-repeat="question in questions">
```
Comment (impguard) [inline app/assets/html/survey_questions_edit.html:5]: Is this really necessary? Shouldn't state knowledge esque behavior take care of this?

File: app/assets/javascripts/startpage/surveyQuestions.coffee [hunk 2]
```diff
@@ -7,19 +7,19 @@ surveyQuestions.config(["$routeProvider", ($routeProvider) ->
         templateUrl: "/assets/survey_questions.html",
         controller: "SurveyQuestionsController"
     })
+    .when("/questions/:id/edit", {
+        templateUrl: "/assets/survey_questions_edit.html",
+        controller: "SurveyQuestionsController"
+    })
```
Comment (impguard) [inline app/assets/javascripts/startpage/surveyQuestions.coffee:13]: Maybe if you can pass in args or use state knowledge features to reuse the normal sureyquestions partial.

File: app/assets/javascripts/startpage/surveySummary.coffee [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    for topicId in StartPageData.getSelectedTopicIds()
+        $scope.selectedTopics[topicId] = allTopics[topicId]
+
+    # The summary text for each responses chosen by the user, grouped by topic id
+    $scope.responseTexts = {}
+    for _id, topic of $scope.selectedTopics
+        questions = StartPageData.getTopicQuestions(topic.id)
+        responseIds = StartPageData.getResponseIdsByTopicId(topic.id)
+        $scope.responseTexts[topic.id] = ""
+        console.log(responseIds)
```
Comment (impguard) [inline app/assets/javascripts/startpage/surveySummary.coffee]: console.log?

File: app/assets/javascripts/startpage/surveyQuestions.coffee [hunk 4]
```diff
@@ -96,13 +96,15 @@ surveyQuestions.config(["$routeProvider", ($routeProvider) ->
...
-        SharedRequests.requestQuestionsByTopic(topicId).success( (allQuestions) ->
-            $scope.questions = []
-            allQuestions = allQuestions.sort((u, v) -> u.index - v.index)
-            for question in allQuestions
-                $scope.questions.push(question)
-            $scope.numQuestions = $scope.questions.length
-        )
+        if $scope.questions.length == 0
+            SharedRequests.requestQuestionsByTopic(topicId).success( (allQuestions) ->
+                $scope.questions = []
```
Comment (whsieh) [inline app/assets/javascripts/startpage/surveyQuestions.coffee:101]: just code style, but you can do $scope.questions = allQuestions.sort (u, v) -> u.index - v.index

File: app/assets/javascripts/startpage/surveySummary.coffee (hunk 3 above)
Comment (leohtkam) [inline app/assets/javascripts/startpage/surveySummary.coffee]: Forgot to remove that ;)

File: app/assets/javascripts/startpage/surveyQuestions.coffee [hunk 5]
```diff
@@ -52,7 +52,7 @@ surveyQuestions.config(["$routeProvider", ($routeProvider) ->
             return "#{questionsLeft} Unanswered Questions"
```
Comment (whsieh) [inline app/assets/javascripts/startpage/surveyQuestions.coffee:52]: what if 1 question is left?

File: app/assets/javascripts/startpage/surveySummary.coffee [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        for question in questions
+            for response in question.survey_responses
+                if responseIds[response.id]
+                    console.log(response)
```
Comment (whsieh) [inline app/assets/javascripts/startpage/surveySummary.coffee]: another console.log ¯_(ツ)_/¯

File: app/assets/html/survey_questions_edit.html (hunk 1 above)
Comment (leohtkam) [inline app/assets/html/survey_questions_edit.html:5]: I will add this to pivotal as a task next iteration to change to using the same page, as that requires using state knowledge.

**Comment by impguard** [comment]
+1 if the other nits are fixed.

**Comment by whsieh** [comment]
+1

**Pull request merged**